### PR TITLE
Upgrade from secure transcript to chat engagement

### PR DESCRIPTION
**Pull request by igorkravchenko**
Add ability to migrate from transcript to chat and continue with regular chat engagement flow.

MOB-1738

### CODE REVIEW

File: GliaWidgets/Sources/ViewController/Chat/ChatViewController.swift [hunk 1]
```diff
@@ -172,3 +172,22 @@ class ChatViewController: EngagementViewController, MediaUpgradePresenter,
...
+extension ChatViewController {
+    func rebind(viewModel: SecureConversations.ChatWithTranscriptModel) {
+        // Binding is only possible for `ChatView` so casting is required.
+        guard let chatView = view as? ChatView else { return }
+        // Swap transcript model with chat model.
+        self.viewModel = viewModel
+        // Swap existing engagement transcript model
+        // (though it is technically not for engagement)
+        // with engagement chat model in superclass EngagementViewController.
+        super.rebind(engagementModel: viewModel.engagementModel)
```
Comment (gersonnoboa) [inline GliaWidgets/Sources/ViewController/Chat/ChatViewController.swift:185]: It looks a bit confusing with `rebind` and then `bind`. What does one do that the other doesn't, and is there a way to describe this in the names?

**Review by EgorovEI — APPROVED** [review]

**Review by rasmustautsglia — APPROVED** [review]
Besides from Gerson comment, all good from me!

**Comment by igorkravchenko** [comment]
!squash

**Comment by igorkravchenko** [comment]
!merge

**Pull request merged**